### PR TITLE
Fixes grinder - CREDITS: GeneralGoose

### PR DESCRIPTION
**Pull request by Gandalf2k15**
Fixes grinder, see https://github.com/HippieStationCode/HippieStation13/pull/2522/files.
Fixes https://github.com/HippieStationCode/HippieStation13/issues/2535

### CODE REVIEW

**Comment by blobbernaut** [comment]
This is a bad way to fix grinders as it reverts a proc that was made literally dozens of times faster and is used everywhere

**Comment by Gandalf2k15** [comment]
it works though.

**Comment by JohnGinnane** [comment]
Can you try to fix it without reverting the PR though?

**Comment by GenericGoose** [comment]
i just found it and suggested to revert it.

**Comment by blobbernaut** [comment]
@RemieRichards help

**Comment by RemieRichards** [comment]
@blobbernaut I actually did the same fix here, except I kept mine and MSO's faster version as a specialist proc (with another proc to assist in building typecaches) and applied it in places that really needed it, like the Tesla, Hostile AI, etc.

**Comment by GenericGoose** [comment]
yeah today i got same idea, to leave new version and re-add old and apply to the broken thing
but really someone's gotta make the new one work

**Comment by GeneralGoose** [comment]
http://puu.sh/pB0tF/dd4686f51d.png
selected tile has no disposal pipe
gg
not like anyone uses security criminal disposals anyway (because it wasnt reported)

**Comment by ghost** [comment]
We should probably merge this after code freeze at the very least.
Sure we won't have the efficient version, but it beats having broken things doesn't it?
We can always refactor the proc back in like Remie has suggested.

File: code/__HELPERS/lists.dm [hunk 1]
```diff
@@ -53,11 +53,10 @@
 
 //Checks for specific types in a list
 /proc/is_type_in_list(atom/A, list/L) //Thank RemieRichards for writing this
```
Comment (JohnGinnane) [inline code/__HELPERS/lists.dm]: I might be nitpicking, but would you mind removing the comment too?

**Comment by JohnGinnane** [comment]
Code looks good. I will merge if comment is remove unless you think it's irrelevant

**Comment by JohnGinnane** [comment]
Code looks good. Going to merge as this is (hopefully) a temporary bug fix. The plan is to implement this again but use the faster proc for heavier things instead, as suggested by Remie

**Pull request merged**